### ISSUE TITLE
Linux on rocket-chip zybo : disk option has no effect

### ISSUE THREAD

**Issue by JishinMaster** [ucb-bar/fpga-zynq#43]
I have a Zybo board loaded with a rocket-chip core (riscv64ima) and I used the linux images obtained with "make fetch-riscv-linux".
I have cross-compiled busybox 1.26.2 and made a new root.bin (removed the old one).
When I use "./fesvr-zynq +disk=/sdcard/riscv/root.bin bbl /sdcard/riscv/vmlinux" or "./fesvr-zynq bbl /sdcard/riscv/vmlinux" it gives me the same result : linux boots and somehow has another filesystem than the one in my root.bin, and contains busybox 1.21.1.

The optional disk option does not seem to have any effect. Since I don't have another root.bin on my sdcard, where does it load busybox from?

**Comment by seldridge** [comment]
I assume that this is coming from an initramfs. The `+disk` option has not been usable for some time. You can specify your own initramfs when you compile your kernel and then use that vmlinux as the payload when you build bbl. Some discussion here which may help: [1](https://github.com/riscv/riscv-pk/issues/18), [2](https://github.com/ucb-bar/fpga-zynq/issues/16).

Chris Celio has a base project that may help with this: https://github.com/ccelio/initramfs_linux_flow. 

However, you only need to run `./fesvr-zynq bbl` to boot linux as everything is packaged inside bbl.

**Comment by davidbiancolin** [comment]
Yes, as @seldridge has suggested, if you are using the provided images, then the bbl instance you are loading onto the target contains an linux image with an initramfs baked in, that will be unpacked and mounted at boot.  @ccelio's base project above is quite useful. 

+disk is not longer supported.  I'm sorry for leaving the README out of date for so long.

**Comment by JishinMaster** [comment]
Thanks for the hints! I didn't manage to get the kernel build use initramfs.txt so I populated the filesystem myself. I used riscv-linux master and soft float seems to work okay on the Zybo.
Thanks again for the support!

Following is what I got : 

root@zynq:~# ./fesvr-zynq bblvmlinux 
              vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
                  vvvvvvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrr       vvvvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrrrrr      vvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrrrrrrr    vvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrrrrrrr    vvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrrrrrrr    vvvvvvvvvvvvvvvvvvvvvvvv
rrrrrrrrrrrrrrrr      vvvvvvvvvvvvvvvvvvvvvv  
rrrrrrrrrrrrr       vvvvvvvvvvvvvvvvvvvvvv    
rr                vvvvvvvvvvvvvvvvvvvvvv      
rr            vvvvvvvvvvvvvvvvvvvvvvvv      rr
rrrr      vvvvvvvvvvvvvvvvvvvvvvvvvv      rrrr
rrrrrr      vvvvvvvvvvvvvvvvvvvvvv      rrrrrr
rrrrrrrr      vvvvvvvvvvvvvvvvvv      rrrrrrrr
rrrrrrrrrr      vvvvvvvvvvvvvv      rrrrrrrrrr
rrrrrrrrrrrr      vvvvvvvvvv      rrrrrrrrrrrr
rrrrrrrrrrrrrr      vvvvvv      rrrrrrrrrrrrrr
rrrrrrrrrrrrrrrr      vv      rrrrrrrrrrrrrrrr
rrrrrrrrrrrrrrrrrr          rrrrrrrrrrrrrrrrrr
rrrrrrrrrrrrrrrrrrrr      rrrrrrrrrrrrrrrrrrrr
rrrrrrrrrrrrrrrrrrrrrr  rrrrrrrrrrrrrrrrrrrrrr

       INSTRUCTION SETS WANT TO BE FREE
[    0.000000] Linux version 4.6.2 (fou@foubuntu) (gcc version 6.1.0 (GCC) ) #36 Sat Mar 18 16:10:52 CET 2017
[    0.000000] Available physical memory: 248MB
[    0.000000] Initial ramdisk at: 0xffffffff80014c48 (2381461 bytes)
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000080800000-0x000000008fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000080800000-0x000000008fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000080800000-0x000000008fffffff]
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 62620
[    0.000000] Kernel command line: 
[    0.000000] PID hash table entries: 1024 (order: 1, 8192 bytes)
[    0.000000] Dentry cache hash table entries: 32768 (order: 6, 262144 bytes)
[    0.000000] Inode-cache hash table entries: 16384 (order: 5, 131072 bytes)
[    0.000000] Sorting __ex_table...
[    0.000000] Memory: 244564K/253952K available (2149K kernel code, 107K rwdata, 564K rodata, 2412K init, 221K bss, 9388K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS:0 nr_irqs:0 0
[    0.000000] clocksource: riscv_clocksource: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 191126044627 ns
[    0.000000] Calibrating delay loop (skipped), value calculated using timer frequency.. 20.00 BogoMIPS (lpj=100000)
[    0.000000] pid_max: default: 32768 minimum: 301
[    0.000000] Mount-cache hash table entries: 512 (order: 0, 4096 bytes)
[    0.000000] Mountpoint-cache hash table entries: 512 (order: 0, 4096 bytes)
[    0.000000] devtmpfs: initialized
[    0.000000] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.000000] NET: Registered protocol family 16
[    0.000000] vgaarb: loaded
[    0.000000] clocksource: Switched to clocksource riscv_clocksource
[    0.000000] NET: Registered protocol family 2
[    0.000000] TCP established hash table entries: 2048 (order: 2, 16384 bytes)
[    0.000000] TCP bind hash table entries: 2048 (order: 2, 16384 bytes)
[    0.000000] TCP: Hash tables configured (established 2048 bind 2048)
[    0.000000] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    0.000000] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    0.000000] NET: Registered protocol family 1
[    0.260000] Unpacking initramfs...
[    0.530000] console [sbi_console0] enabled
[    0.530000] futex hash table entries: 256 (order: 0, 6144 bytes)
[    0.540000] io scheduler noop registered
[    0.540000] io scheduler cfq registered (default)
[    0.600000] Freeing unused kernel memory: 2412K (ffffffff80000000 - ffffffff8025b000)
[    0.600000] This architecture does not have kernel memory protection.
/ # source /etc/init.d/rcS 
mount: mounting none on /proc failed: Device or resource busy
/ # busybox
BusyBox v1.26.2 (2017-03-18 15:57:17 CET) multi-call binary.
BusyBox is copyrighted by many authors between 1998-2015.
Licensed under GPLv2. See source distribution for detailed
copyright notices.

Usage: busybox [function [arguments]...]
   or: busybox --list[-full]
   or: busybox --install [-s] [DIR]
   or: function [arguments]...

	BusyBox is a multi-call binary that combines many common Unix
	utilities into a single executable.  Most people will create a
	link to busybox for each function they wish to use and BusyBox
	will act like whatever it was invoked as.

Currently defined functions:
	[, [[, acpid, add-shell, addgroup, adduser, adjtimex, arp, arping, ash,
	awk, base64, basename, beep, blkdiscard, blkid, blockdev, bootchartd,
	brctl, bunzip2, bzcat, bzip2, cal, cat, catv, chat, chattr, chgrp,
	chmod, chown, chpasswd, chpst, chroot, chrt, chvt, cksum, clear, cmp,
	comm, conspy, cp, cpio, crond, crontab, cryptpw, cttyhack, cut, date,
	dc, dd, deallocvt, delgroup, deluser, depmod, devmem, df, dhcprelay,
	diff, dirname, dmesg, dnsd, dnsdomainname, dos2unix, dpkg, dpkg-deb,
	du, dumpkmap, dumpleases, echo, ed, egrep, eject, env, envdir,
	envuidgid, ether-wake, expand, expr, fakeidentd, false, fatattr, fbset,
	fbsplash, fdflush, fdformat, fdisk, fgconsole, fgrep, find, findfs,
	flock, fold, free, freeramdisk, fsck, fsck.minix, fstrim, fsync, ftpd,
	ftpget, ftpput, fuser, getopt, getty, grep, groups, gunzip, gzip, halt,
	hd, hdparm, head, hexdump, hostid, hostname, httpd, hush, hwclock,
	i2cdetect, i2cdump, i2cget, i2cset, id, ifconfig, ifdown, ifenslave,
	ifplugd, ifup, inetd, init, insmod, install, ionice, iostat, ip,
	ipaddr, ipcalc, ipcrm, ipcs, iplink, ipneigh, iproute, iprule,
	iptunnel, kbd_mode, kill, killall, killall5, klogd, last, less,
	linux32, linux64, linuxrc, ln, loadfont, loadkmap, logger, login,
	logname, logread, losetup, lpd, lpq, lpr, ls, lsattr, lsmod, lsof,
	lspci, lsusb, lzcat, lzma, lzop, lzopcat, makedevs, makemime, man,
	md5sum, mdev, mesg, microcom, mkdir, mkdosfs, mke2fs, mkfifo,
	mkfs.ext2, mkfs.minix, mkfs.vfat, mknod, mkpasswd, mkswap, mktemp,
	modinfo, modprobe, more, mount, mountpoint, mpstat, mt, mv, nameif,
	nanddump, nandwrite, nbd-client, nc, netstat, nice, nmeter, nohup,
	nsenter, nslookup, ntpd, od, openvt, passwd, patch, pgrep, pidof, ping,
	ping6, pipe_progress, pivot_root, pkill, pmap, popmaildir, poweroff,
	powertop, printenv, printf, ps, pscan, pstree, pwd, pwdx, raidautorun,
	rdate, rdev, readahead, readlink, readprofile, realpath, reboot,
	reformime, remove-shell, renice, reset, resize, rev, rm, rmdir, rmmod,
	route, rpm, rpm2cpio, rtcwake, run-parts, runlevel, runsv, runsvdir,
	rx, script, scriptreplay, sed, sendmail, seq, setarch, setconsole,
	setfont, setkeycodes, setlogcons, setserial, setsid, setuidgid, sh,
	sha1sum, sha256sum, sha3sum, sha512sum, showkey, shuf, slattach, sleep,
	smemcap, softlimit, sort, split, start-stop-daemon, stat, strings,
	stty, su, sulogin, sum, sv, svc, svlogd, swapoff, swapon, switch_root,
	sync, sysctl, syslogd, tac, tail, tar, tcpsvd, tee, telnet, telnetd,
	test, tftp, tftpd, time, timeout, top, touch, tr, traceroute,
	traceroute6, true, truncate, tty, ttysize, tunctl, ubiattach,
	ubidetach, ubimkvol, ubirename, ubirmvol, ubirsvol, ubiupdatevol,
	udhcpc, udhcpd, udpsvd, uevent, umount, uname, unexpand, uniq,
	unix2dos, unlink, unlzma, unlzop, unshare, unxz, unzip, uptime, users,
	usleep, uudecode, uuencode, vconfig, vi, vlock, volname, wall, watch,
	watchdog, wc, wget, which, who, whoami, whois, xargs, xz, xzcat, yes,
	zcat, zcip
/ # ls /bin
ash            dumpkmap       kbd_mode       mv             setarch
base64         echo           kill           netstat        setserial
busybox        ed             linux32        nice           sh
cat            egrep          linux64        pidof          sleep
catv           false          ln             ping           stat
chattr         fatattr        login          ping6          stty
chgrp          fdflush        ls             pipe_progress  su
chmod          fgrep          lsattr         printenv       sync
chown          fsync          lzop           ps             tar
conspy         getopt         makemime       pwd            touch
cp             grep           mkdir          reformime      true
cpio           gunzip         mknod          rev            umount
cttyhack       gzip           mktemp         rm             uname
date           hostname       more           rmdir          usleep
dd             hush           mount          rpm            vi
df             ionice         mountpoint     run-parts      watch
dmesg          iostat         mpstat         scriptreplay   zcat
dnsdomainname  ipcalc         mt             sed

`
